lib/class/json_data.class.php

```diff
@@ -0,0 +1,638 @@
+ *
+ * LICENSE: GNU Affero General Public License, version 3 (AGPLv3)
+ * Copyright 2001 - 2016 Ampache.org
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * JSON_Data Class
+ *
+ * This class takes care of all of the JSON document stuff in Ampache these
+ * are all static calls
+ *
+ */
+class JSON_Data
+{
+    // This is added so that we don't pop any webservers
+    private static $limit  = 5000;
+    private static $offset = 0;
+    private static $type   = '';
+
+    /**
+     * constructor
+     *
+     * We don't use this, as its really a static class
+     */
+    private function __construct()
+    {
+        // Rien a faire
+    } // constructor
+
+    /**
+     * set_offset
+     *
+     * This takes an int and changes the offset
+     *
+     * @param    integer    $offset    (description here...)
+     * @return    void
+     */
+    public static function set_offset($offset)
+    {
+        $offset       = intval($offset);
+        self::$offset = $offset;
+    } // set_offset
+
+    /**
+     * set_limit
+     *
+     * This sets the limit for any ampache transactions
+     *
+     * @param    integer    $limit    (description here...)
+     * @return    void
+     */
+    public static function set_limit($limit)
+    {
+        if (!$limit) {
+            return false;
+        }
+
+        $limit       = intval($limit);
+        self::$limit = $limit;
+    } // set_limit
+
+    /**
+     * error
+     *
+     * This generates a JSON Error message
+     * nothing fancy here...
+     *
+     * @param    integer    $code    Error code
+     * @param    string    $string    Error message
+     * @return    string    return error message JSON
+     */
+    public static function error($code,$string)
+    {
+        $JSON = json_encode(array("error" => array("code" => $code, "message" => $string)), JSON_PRETTY_PRINT);
+        return $JSON;
+    } // error
+
+    /**
+     * single_string
+     *
+     * This takes two values, first the key second the string
+     *
+     * @param    string    $key    (description here...)
+     * @param    string    $string    JSON data
+     * @return    string    return JSON
+     */
+    public static function single_string($key, $string='')
+    {
+        if (!empty($string)) {
+            $JSON = json_encode(array($key => $string), JSON_PRETTY_PRINT);
```

Comment:
I think `json_encode` calls in this method should go in a `_encode` private method, which will avoid code duplication and ease change of options such as `JSON_PRETTY_PRINT` across all the API responses.